simd/src/Kokkos_SIMD_NEON.hpp

```diff
@@ -0,0 +1,964 @@
+  operator||(neon_mask const& other) const {
+    return Derived(vorrq_u64(m_value, other.m_value));
+  }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION Derived
+  operator&&(neon_mask const& other) const {
+    return Derived(vandq_u64(m_value, other.m_value));
+  }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION Derived operator!() const {
+    auto const true_value = static_cast<uint64x2_t>(neon_mask(true));
+    return Derived(veorq_u64(m_value, true_value));
+  }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION bool operator==(
+      neon_mask const& other) const {
+    uint64x2_t const elementwise_equality = vceqq_u64(m_value, other.m_value);
+    uint32x2_t const narrow_elementwise_equality =
+        vqmovn_u64(elementwise_equality);
+    uint64x1_t const overall_equality_neon =
+        vreinterpret_u64_u32(narrow_elementwise_equality);
+    uint64_t const overall_equality = vget_lane_u64(overall_equality_neon, 0);
+    return overall_equality == 0xFFFFFFFFFFFFFFFFULL;
+  }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION bool operator!=(
+      neon_mask const& other) const {
+    return !operator==(other);
+  }
+};
+
+template <class Derived>
+class neon_mask<Derived, 32> {
+  uint32x2_t m_value;
+
+ public:
+  class reference {
+    uint32x2_t& m_mask;
+    int m_lane;
+
+   public:
+    KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION reference(uint32x2_t& mask_arg,
+                                                    int lane_arg)
+        : m_mask(mask_arg), m_lane(lane_arg) {}
+    KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION reference
+    operator=(bool value) const {
+      switch (m_lane) {
+        case 0:
+          m_mask = vset_lane_u32(value ? 0xFFFFFFFFU : 0, m_mask, 0);
+          break;
+        case 1:
+          m_mask = vset_lane_u32(value ? 0xFFFFFFFFU : 0, m_mask, 1);
+          break;
+      }
+      return *this;
+    }
+    KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION operator bool() const {
+      switch (m_lane) {
+        case 0: return vget_lane_u32(m_mask, 0) != 0;
+        case 1: return vget_lane_u32(m_mask, 1) != 0;
+      }
+      return false;
+    }
+  };
+  using value_type          = bool;
+  using abi_type            = simd_abi::neon_fixed_size<2>;
+  using implementation_type = uint32x2_t;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION neon_mask() = default;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION explicit neon_mask(value_type value)
+      : m_value(vdup_n_u32(value ? 0xFFFFFFFFU : 0)) {}
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION static constexpr std::size_t size() {
+    return 2;
+  }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION constexpr explicit neon_mask(
+      uint32x2_t const& value_in)
+      : m_value(value_in) {}
+  template <class U>
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION neon_mask(neon_mask<U, 64> const& other)
+      : m_value(vqmovn_u64(static_cast<uint64x2_t>(other))) {}
+  template <class U>
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION neon_mask(neon_mask<U, 32> const& other)
+      : m_value(static_cast<uint32x2_t>(other)) {}
```

Comment:
Please move these two definition just above `size()`.
It gives friendlier diff.